### PR TITLE
add BCP 47 Language Codes

### PR DESCRIPTION
**Pull request by dericed**

### CODE REVIEW

File: ebml_matroska.xml [hunk 1]
```diff
@@ -245,7 +245,10 @@
     <documentation lang="en">A human-readable track name.</documentation>
   </element>
   <element name="Language" path="0*1(\Segment\Tracks\TrackEntry\Language)" cppname="TrackLanguage" id="0x22B59C" type="string" maxOccurs="1" minver="1" default="eng">
-    <documentation lang="en">Specifies the language of the track in the <a href="http://www.matroska.org/technical/specs/index.html#languages">Matroska languages form</a>.</documentation>
+    <documentation lang="en">Specifies the language of the track in the <a href="http://www.matroska.org/technical/specs/index.html#languages">Matroska languages form</a>. This Element MUST be ignored if the LanguageIETF Element is used in the same TrackEntry.</documentation>
```
Comment (robUx4) [inline ebml_matroska.xml]: backtick `LanguageIETF`

Comment (robUx4) [inline ebml_matroska.xml]: Or use a link to the element

File: ebml_matroska.xml [hunk 2]
```diff
@@ -827,7 +830,10 @@
     <documentation lang="en">Contains the string to use as the chapter atom.</documentation>
   </element>
   <element name="ChapLanguage" path="1*(\Segment\Chapters\EditionEntry\ChapterAtom\ChapterDisplay\ChapLanguage)" cppname="ChapterLanguage" id="0x437C" type="string" minOccurs="1" minver="1" webm="1" default="eng">
-    <documentation lang="en">The languages corresponding to the string, in the <a href="http://www.loc.gov/standards/iso639-2/php/English_list.php">bibliographic ISO-639-2 form</a>.</documentation>
+    <documentation lang="en">The languages corresponding to the string, in the <a href="http://www.loc.gov/standards/iso639-2/php/English_list.php">bibliographic ISO-639-2 form</a>. This Element MUST be ignored if the ChapLanguageIETF Element is used within the same ChapterDisplay Element.</documentation>
```
Comment (robUx4) [inline ebml_matroska.xml]: Backtick or add a link to `ChapLanguageIETF`

File: ebml_matroska.xml [hunk 3]
```diff
@@ -889,7 +895,10 @@
     <documentation lang="en">The name of the Tag that is going to be stored.</documentation>
   </element>
   <element name="TagLanguage" path="1*1(\Segment\Tags\Tag\SimpleTag\TagLanguage)" id="0x447A" type="string" minOccurs="1" maxOccurs="1" minver="1" webm="0" default="und">
-    <documentation lang="en">Specifies the language of the tag specified, in the <a href="http://www.matroska.org/technical/specs/index.html#languages">Matroska languages form</a>.</documentation>
+    <documentation lang="en">Specifies the language of the tag specified, in the <a href="http://www.matroska.org/technical/specs/index.html#languages">Matroska languages form</a>. This Element MUST be ignored if the TagLanguageIETF Element is used within the same SimpleTag Element.</documentation>
```
Comment (robUx4) [inline ebml_matroska.xml]: Backtick or add a link to `TagLanguageIETF`

File: index.md [hunk 4]
```diff
@@ -71,7 +71,11 @@ All top-levels elements (Segment and direct sub-elements) are coded on 4 octets,
...
+Matroska from version 1 through 3 uses language codes that can be either the 3 letters [bibliographic ISO-639-2](http://www.loc.gov/standards/iso639-2/php/English_list.php) form (like "fre" for french), or such a language code followed by a dash and a country code for specialities in languages (like "fre-ca" for Canadian French). The `ISO 639-2 Language Elements` are "Language Element", "TagLangauge Element", and "ChapLanguage Element".
+
+Starting in Matroska version 4, either `ISO 639-2` or [BCP 47](https://tools.ietf.org/html/bcp47) MAY be used, although `BCP 47` is RECOMMENDED. The `BCP 47 Language Elements` are "LanguageIETF Element", "TagLangaugeIETF Element", and "ChapLanguageIETF Element". If a `BCP 47 Language Element` and an `ISO 639-2 Language Element` are used within the same `Parent Element` then the `ISO 639-2 Lanaguage Element` MUST be ignored and precedence given to the `BCP 47 Language Element`.
```
Comment (robUx4) [inline index.md]: TagLangaugeIETF typo

File: ebml_matroska.xml (hunk 1 above)
Comment (dericed) [inline ebml_matroska.xml]: hmm we haven't done backticked in XML yet

File: index.md (hunk 4 above)
Comment (dericed) [inline index.md]: fixed and forced updated

File: ebml_matroska.xml (hunk 1 above)
Comment (dericed) [inline ebml_matroska.xml]: I suggest the backticking and inter-element linking to be resolved in a later PR, as it is needed throughout and requires some negotiation between markdown, xml and rfc handling.

Comment (robUx4) [inline ebml_matroska.xml]: ok

**Comment by robUx4** [comment]
LGTM

**Comment by retokromer** [comment]
LGTM

File: ebml_matroska.xml [hunk 5]
```diff
@@ -245,7 +245,10 @@
     <documentation lang="en">A human-readable track name.</documentation>
   </element>
   <element name="Language" path="0*1(\Segment\Tracks\TrackEntry\Language)" cppname="TrackLanguage" id="0x22B59C" type="string" maxOccurs="1" minver="1" default="eng">
-    <documentation lang="en">Specifies the language of the track in the <a href="http://www.matroska.org/technical/specs/index.html#languages">Matroska languages form</a>.</documentation>
+    <documentation lang="en">Specifies the language of the track in the <a href="http://www.matroska.org/technical/specs/index.html#languages">Matroska languages form</a>. This Element MUST be ignored if the LanguageIETF Element is used in the same TrackEntry.</documentation>
+  </element>
+  <element name="LanguageIETF" path="0*1(\Segment\Tracks\TrackEntry\LanguageIETF)" id="0x22B59D" type="string" maxOccurs="1" minver="4">
+    <documentation lang="en">Specifies the language of the track according to <a href="https://tools.ietf.org/html/bcp47">BCP 47</a> and using the <a href="http://www.iana.com/assignments/language-subtag-registry/language-subtag-registry">IANA Language Subtag Registry</a>.</documentation>
```
Comment (mbunkus) [inline ebml_matroska.xml]: Maybe mention here that the presence of `LanguageIETF` means that `Language` must be ignored, too, not just with the `Language` attribute. Same for the other places where `XYZLanguageIETF` is defined.

File: index.md [hunk 6]
```diff
@@ -71,7 +71,11 @@ All top-levels elements (Segment and direct sub-elements) are coded on 4 octets,
...
+Matroska from version 1 through 3 uses language codes that can be either the 3 letters [bibliographic ISO-639-2](http://www.loc.gov/standards/iso639-2/php/English_list.php) form (like "fre" for french), or such a language code followed by a dash and a country code for specialities in languages (like "fre-ca" for Canadian French). The `ISO 639-2 Language Elements` are "Language Element", "TagLanguage Element", and "ChapLanguage Element".
+
+Starting in Matroska version 4, either `ISO 639-2` or [BCP 47](https://tools.ietf.org/html/bcp47) MAY be used, although `BCP 47` is RECOMMENDED. The `BCP 47 Language Elements` are "LanguageIETF Element", "TagLanguageIETF Element", and "ChapLanguageIETF Element". If a `BCP 47 Language Element` and an `ISO 639-2 Language Element` are used within the same `Parent Element` then the `ISO 639-2 Lanaguage Element` MUST be ignored and precedence given to the `BCP 47 Language Element`.
```
Comment (mbunkus) [inline index.md]: A missing `,` between `Parent Element` and `then`.

Comment (dericed) [inline index.md]: added

File: ebml_matroska.xml (hunk 5 above)
Comment (dericed) [inline ebml_matroska.xml]: I added a version of `If this Element is used, then any XYZLanguage Elements used in the same TrackEntry|ChapterDisplay|SimpleTag MUST be ignored.` to each XYZLanguageIETF Element.

**Comment by dericed** [comment]
I updated the PR to respond to @mbunkus's suggestion. I've still left defined terms without backticks and suggest handling that in a later PR.

File: ebml_matroska.xml (hunk 3 above)
Comment (retokromer) [inline ebml_matroska.xml]: `https://`

File: ebml_matroska.xml [hunk 7]
```diff
@@ -827,10 +830,13 @@
...
+    <documentation lang="en">Specifies the language used in the ChapString according to <a href="https://tools.ietf.org/html/bcp47">BCP 47</a> and using the <a href="http://www.iana.com/assignments/language-subtag-registry/language-subtag-registry">IANA Language Subtag Registry</a>. If this Element is used, then any ChapLanguage Elements used in the same ChapterDisplay MUST be ignored.</documentation>
   </element>
   <element name="ChapCountry" path="0*(\Segment\Chapters\EditionEntry\ChapterAtom\ChapterDisplay\ChapCountry)" cppname="ChapterCountry" id="0x437E" type="string" minver="1" webm="0">
-    <documentation lang="en">The countries corresponding to the string, same 2 octets as in <a href="http://www.iana.org/cctld/cctld-whois.htm">Internet domains</a>.</documentation>
+    <documentation lang="en">The countries corresponding to the string, same 2 octets as in <a href="http://www.iana.org/cctld/cctld-whois.htm">Internet domains</a>. This Element MUST be ignored if the ChapLanguageIETF Element is used within the same ChapterDisplay Element.</documentation>
```
Comment (retokromer) [inline ebml_matroska.xml]: `https://`

File: ebml_matroska.xml (hunk 2 above)
Comment (retokromer) [inline ebml_matroska.xml]: gives a `Page Not Found` error

File: ebml_matroska.xml (hunk 7 above)
Comment (dericed) [inline ebml_matroska.xml]: done

File: index.md [hunk 8]
```diff
@@ -71,7 +71,11 @@ All top-levels elements (Segment and direct sub-elements) are coded on 4 octets,
 
 ### Language Codes
 
-Language codes can be either the 3 letters [bibliographic ISO-639-2](http://www.loc.gov/standards/iso639-2/php/English_list.php) form (like "fre" for french), or such a language code followed by a dash and a country code for specialities in languages (like "fre-ca" for Canadian French). Country codes are the same as used for [internet domains](http://www.iana.org/cctld/cctld-whois.htm).
+Matroska from version 1 through 3 uses language codes that can be either the 3 letters [bibliographic ISO-639-2](http://www.loc.gov/standards/iso639-2/php/English_list.php) form (like "fre" for french), or such a language code followed by a dash and a country code for specialities in languages (like "fre-ca" for Canadian French). The `ISO 639-2 Language Elements` are "Language Element", "TagLanguage Element", and "ChapLanguage Element".
```
Comment (retokromer) [inline index.md]: `https://`

**Comment by robUx4** [comment]
LGTM.

IMO the `https;//` changes are welcome but should be in another issue/PR.

**Comment by retokromer** [comment]
@robUx4 OK, will make a specific PR.

**Pull request merged**